frontend/ui/widget/touchmenu.lua

```diff
@@ -961,4 +975,227 @@ function TouchMenu:onBack()
+                end
+            end
+        end
+
+        if val.sub_item_table_func and extensive_search then
+            local sub_item_table = val.sub_item_table_func()
+            local perpage = ""
+            if sub_item_table.max_per_page then
+                perpage = "[" .. sub_item_table.max_per_page .."]"
+            end
+            recurse(sub_item_table, path .. ".sub_item_table_func" .. perpage , text, depth)
+        elseif val.sub_item_table then
+            local perpage = ""
+            if val.sub_item_table.max_per_page then
+                perpage = "[" .. val.sub_item_table.max_per_page .."]"
+            end
+            recurse(val.sub_item_table, path .. ".sub_item_table" .. perpage , text, depth)
+        end
+    end -- recurse
+
+    -- initial call of recurse
+    for i = 1, #self.tab_item_table do
+        recurse(self.tab_item_table[i], i, self.tab_item_table[i].text or "ROOT"..i, 0)
+    end
+
+--[[
+    for i = 1, #found_menu_items do
+        print("xxxxxxx->", i,
+            found_menu_items[i][1],
+            found_menu_items[i][2],
+            found_menu_items[i][3])
+    end
+]]
+    return found_menu_items
+end
+
+function TouchMenu:openMenu(path)
+    -- first switch to correct MenuTab
+    local sep_pos = path:find("%.")
+    local tab_num = tonumber(path:sub(1, sep_pos - 1))
+    path = path:sub(sep_pos + 1)
+    local item = self.tab_item_table[tab_num]
+
+    if not tab_num then
+        return
+    end
+
+    self:switchMenuTab(tab_num)
+    self.bar:switchToTab(tab_num)
+    local dummy, num_of_sep = path:gsub("%.", "%.")
+    while num_of_sep > 1 do
+        sep_pos = path:find("%.")
+        local identifier = path:sub(1, sep_pos -1)
+        local item_num = tonumber(identifier)
+        path = path:sub(sep_pos + 1)
+        if item_num then
+            item = item[item_num]
+            self:onMenuSelect(item)
+        elseif identifier:find("sub_item_table_func") then
+            item = item.sub_item_table_func()
+        elseif identifier:find("sub_item_table") then
+            item = item.sub_item_table
+        end
+        num_of_sep = num_of_sep - 1
+    end
+
+    -- Now we are in the right menu, but maybe in the wrong page
+    sep_pos = path:find("%.")
+    if not sep_pos then
+        local logger = require("logger")
+        logger.err("TouchMenu: search; internal error") -- should not happen
+        return
+    end
+    local identifier = path:sub(sep_pos + 1)
+    local item_num = tonumber(identifier)
+
+    local perpage
+    if path:find("sub_item_table_func%[") then
+        perpage = path:sub(#("sub_item_table_func%["), sep_pos - 2)
+    elseif path:find("sub_item_table%[") then
+        perpage = path:sub(#("sub_item_table%["), sep_pos - 2)
+    end
+    perpage = tonumber(perpage) or self.perpage
+
+    local page_num = math.ceil(item_num / perpage)
+    UIManager:nextTick(self.onPage, self, page_num)
```

Comment:
Feels cleaner to just have an anonymous function here - like we do everywhere - instead of a callable + self + arg.